composables/layout.ts

```diff
@@ -0,0 +1,31 @@
+export const useSmallScreen = () => useState<boolean | undefined>(
+  'small-screen',
+  () => false
+)
+
+export const useNavDrawerOpen = () => useState<boolean | undefined>(
+  'nav-drawer-open',
+  () => true
+)
+
+const smallScreenWidth = 800
+
+const smallScreen = useSmallScreen()
+const navDrawerOpen = useNavDrawerOpen()
+
+if (window.innerWidth < smallScreenWidth) {
+  smallScreen.value = true
+  navDrawerOpen.value = false
+}
+
+window.addEventListener('resize', () => {
+  if (window.innerWidth < smallScreenWidth) smallScreen.value = true
+  else {
+    smallScreen.value = false
+    if (!navDrawerOpen.value) navDrawerOpen.value = true
+  }
+}, true)
+
+export const toggleNavDrawer = () => {
+  navDrawerOpen.value = !navDrawerOpen.value
+}
```

Comment:
Newline at end of files plz

Reply:
Idk if this is the way composables are meant to be used in nuxt.  Like, it works, but I think composables are meant to be small, be about one thing, and named after that one thing.  e.g. `composables/signer.ts` becomes `useSigner`

Reply:
I'm using them like useContext / hooks in react. Global state than can be accessed anywhere in the tree + any crud operations for that state included in the file is typically the best practice. So like those crud functions can also be accessed anywhere as well

Reply:
Or like any kind of state store in also Angular. Encapsulate all the update logic / functions within the store as well

Reply:
ye, that is one of the main use cases, what i'm talking about is how nuxt hooks them up automatically. i guess it's fine cause it works lol